### PR TITLE
Add integration tests for Tags API

### PR DESCRIPTION
**Pull request by riteshjagga**
Please review the integration tests of Tags API

### CODE REVIEW

File: spec/factories/tag.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :tag do
+    name {Faker::Lorem.characters(10)}
```
Comment (riteshjagga) [inline spec/factories/tag.rb:3]: Factory to create tag with an arbitrary name

File: spec/models/tag_spec.rb [hunk 2]
```diff
@@ -4,5 +4,7 @@
 
   #Validation tests
   it {should validate_presence_of(:name)}
+  it {should validate_uniqueness_of(:name)}
+  it {should validate_length_of(:name).is_at_least(3).is_at_most(25)}
```
Comment (riteshjagga) [inline spec/models/tag_spec.rb:8]: Model test cases

File: spec/requests/tags_spec.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+RSpec.describe 'Tag API', type: :request do
```
Comment (riteshjagga) [inline spec/requests/tags_spec.rb:3]: This file contains integration tests of Tags API

File: spec/support/api/schemas/tag.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (riteshjagga) [inline spec/support/api/schemas/tag.json:1]: **Gem: `json-schema`**

This JSON Schema is used to validate the data structure of a tag returned in the GET tag API.

**Review by riteshjagga — COMMENTED** [review]
Please review the integration tests of Tags API

File: spec/requests/tags_spec.rb [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      get "/tags/#{tag_id}"
+      json = JSON.parse(response.body)
+      puts "json = #{json}"
+      expect(response).to match_response_schema("tag")
+    end
+  end
+
+  describe 'GET /tags/:name/todos' do
+    it 'gets todos associated with tag name' do
+      todo_1 = tag.todos.create({title: "Todo Item 1"})
```
Comment (2018-freecoder) [inline spec/requests/tags_spec.rb:40]: when write long test, there is a good practice to separate test into 3 parts:
1. preparation, such as create data
2. perform action, such as send request, or invoke method
3. assert result

there is a empty line between each parts, and using this practice, I think it is good for other people read and understand your long test :)

**Review by 2018-freecoder — APPROVED** [review]

**Pull request merged**